sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -111,24 +113,14 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
    * @throws Error if an open receiver is already existing for given sessionId.
    */
   private constructor(
-    context: ClientEntityContext,
+    private _messageSession: MessageSession,
+    private _context: ClientEntityContext,
     public receiveMode: "peekLock" | "receiveAndDelete",
-    private _sessionOptions: CreateSessionReceiverOptions,
-    retryOptions: RetryOptions = {}
+    private _retryOptions: RetryOptions = {}
   ) {
-    throwErrorIfConnectionClosed(context.namespace);
-    this._context = context;
+    throwErrorIfConnectionClosed(this._context.namespace);
     this.entityPath = this._context.entityPath;
-    this._retryOptions = retryOptions;
-
-    if (this._sessionOptions.sessionId) {
-      this._sessionOptions.sessionId = String(this._sessionOptions.sessionId);
-    }
-
-    // `createInitializedSessionReceiver` will set this value by calling init()
-    // so we just temporarily set it to "" so we can get away with it never being
-    // `undefined`.
-    this.sessionId = "";
+    this.sessionId = _messageSession.sessionId!;
```

Comment:
One thing I don't like about the MessageSession object is that you're forced to just assume that sessionId is set in classes external to it.

You can actually declare the `sessionId` field in MessageSession like this:

```typescript
class MessageSession {
  sessionId!: string;  
}
```

Which will then put the responsibility of the string not being null/undefined in the right spot (on MessageSession, rather than consumers of MessageSession). Since we only allow construction via the static `create()` method that seems better to me.

Reply:
Hmmmm.. we are using the same variable for both the user-provided sessionId and the receivedSessionId.
Decided to split it up into `providedSessionId` which can be optional and the `sessionId` which must be a string.

```ts
  /**
   * @property {string} [sessionId] The sessionId provided in the MessageSessionOptions. Empty string is valid sessionId.
   */
  providedSessionId?: string;
  /**	
   * @property {string} [sessionId] The sessionId for the message session. Empty string is valid sessionId
   */
  sessionId!: string;
```
And thus removed even more type annoyances in MessageSession - https://github.com/Azure/azure-sdk-for-js/pull/8849/commits/0c0a0a35a7152e2613656b026351ef4197144ee7

Let me know if this looks good or if there are any other concerns.

Reply:
> Which will then put the responsibility of the string not being null/undefined in the right spot (on MessageSession, rather than consumers of MessageSession). Since we only allow construction via the static create() method that seems better to me.

One thing to note is that the `MessageSession.sessionId` is not guaranteed to be a non undefined/null value in the constructor of the `MessageSession` class. It is only when `MessageSession.init()` is called successfully, one can expect `MessageSession.sessionId` to have a valid value.

So, adding `!` to the property definition in `MessageSession` can be misleading.

Reply:
What I meant to say was that creators of MessageSession are still responsible to ensure that the sessionId on it is valid by calling init()

Reply:
I'm confused now, should I revert?

Reply:
Lets leave as is based on offline discussions